### PR TITLE
Semi-completed The Submit DiscussionBoard/Message Use Case.

### PR DESCRIPTION
**Pull request by HumrajB**
There are still modifications that need to be made that concern other members such as the TestDocument Class and SolutionDocument Class (@N1cholasMcKee) which is why there is still an error in the interactor. I also made my own tree class in the entities folder to store messages which might need to be moved to a different folder as it is not an entity. I also modified the SolutionDocument Class as there is no longer a Disscussion Board Class.

### CODE REVIEW

**Comment by HumrajB** [comment]
I also still have to do javadoc etc.

**Review by HarveyRDonnelly — DISMISSED** [review]
This is really great work Humraj! The Tree class is very well designed with quality recursive methods. I think it is appropriate to consider the Tree class an entity but perhaps we can rename it to show that it is a message tree specifically (e.g. "MessageTree"). Compilation issues can be resolved in a future PR as we integrate software components. JavaDoc can also be added in future PR.

File: src/main/java/uc/dboard/submessage/SubmitDBMessageInteractor.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        Course course = stateTracker.getCourseIfTracked(courseId);
+
+        //Below Code Traverses through Arraylist to find the test that corresponds to where the message should be added
+
+        //Course should have a tests attribute that is an ArrayList
+        ArrayList<TestDocument> tests = course.tests;
+        TestDocument relatedTest = null;
+        for (TestDocument test:tests){
+            if (test.getId().equals(testId)){
+                relatedTest = test;
```
Comment (HarveyRDonnelly) [inline src/main/java/uc/dboard/submessage/SubmitDBMessageInteractor.java:42]: I think the Course entity will have a dictionary mapping the test ID to the test document object. This code can be updated once that's implemented.

**Review by HarveyRDonnelly — APPROVED** [review]
Merge conflicts resolved re: recordedTime variable.

**Pull request merged**